lib/components/logIn.dart

```diff
@@ -372,63 +321,62 @@ class _LogInScreenState extends State<LogInScreen> {
+                      height: size.height*0.01,
+                    ),
+                    Container(
+                      width: MediaQuery.of(context).size.width - 50,
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.center,
+                        children: [
+                          Expanded(
+                              child: Divider(
+                            color: Colors.grey,
+                          )),
+                          Text(
+                            "Or Connect",
+                            style: TextStyle(color: Colors.grey),
+                          ),
+                          Expanded(
+                              child: Divider(
+                            color: Colors.grey,
+                          )),
+                        ],
                       ),
-                      Expanded(
-                          child: Divider(
-                        color: Colors.grey,
-                      )),
-                    ],
-                  ),
-                ),
-                SizedBox(height: 15),
-                Row(
-                  mainAxisAlignment: MainAxisAlignment.center,
-                  children: [
-                    GoogleAuthButton(
-                        borderRadius: 29,
-                        iconSize: 30,
-                        style: AuthButtonStyle.icon,
-                        onPressed: () async {
-                          await _auth.signInWithGoogle().then(
-                            (result) async {
-                              if (result != null) {
-                                _image = await urlToFile(_auth.imageUrl);
-                                await uploadProfilePic(context);
-                                Navigator.pushAndRemoveUntil(context,
-                                    MaterialPageRoute(
-                                  builder: (context) {
-                                    return HomeScreen(_auth.userName);
-                                  },
-                                ), (route) => false);
-                              }
-                            },
-                          );
-                        }),
+                    ),
+                    SizedBox(height: size.height*0.01),
+                    Row(mainAxisAlignment: MainAxisAlignment.center, children: [
+                      GoogleAuthButton(
+                          borderRadius: 29,
+                          iconSize: 30,
+                          style: AuthButtonStyle.icon,
+                          onPressed: () async {
+                            var _auth = AuthService();
+                            final result = await _auth.signInWithGoogle();
+                            if (result != null) {
+                              _image = await urlToFile(_auth.imageUrl);
+                              var firebase = FirebaseStorageServices(_image, context);
```

Comment:
Ahh, this is all the wrong way to do it, you're still creating a new instance of this class every time you hit on `onPress`.